js/modules/k6/experimental/streams/readable_streams.go

```diff
@@ -0,0 +1,586 @@
+package streams
+
+import (
+	"errors"
+
+	"github.com/dop251/goja"
+	"go.k6.io/k6/js/common"
+	"go.k6.io/k6/js/modules"
+	"go.k6.io/k6/js/promises"
+)
+
+// ReadableStream is a concrete instance of the general [readable stream] concept.
+//
+// It is adaptable to any chunk type, and maintains an internal queue to keep track of
+// data supplied by the underlying source but not yet read by any consumer.
+//
+// [readable stream]: https://streams.spec.whatwg.org/#rs-class
+type ReadableStream struct {
+	// Locked indicate whether the readable stream is locked to a reader
+	Locked bool
+
```

Comment:
I think the explanation is partly similar to https://github.com/grafana/k6/pull/3696#discussion_r1579132598.

To satisfy some of the edge cases verified by WPTs, like asserting the property exists, it's read-only, it's a boolean, etc, etc; plus because at least for now we don't _manually_ "convert" or "transform" the Go object to a `*goja.Object`, as we do with `ReadableStreamDefaultController`.

I guess in the future that might be the way to go, but unless there's something that's really blocking, I'd leave to that for a later refactor, especially considering that this as-is is (_or seems to be_) WPTs compliant.